### PR TITLE
The types now have the ability to be sorted via a pluggable Comparators

### PR DESCRIPTION
**Pull request by bbakerman**
relates to #1534 

The types now have the ability to be sorted via a pluggable Comparators and now present themselves BY default in "as is" order

Since all of the maps / lists are now LinkedHashMap the insertion order will be stable.

People can then use a pluggable `graphql.schema.GraphqlTypeComparatorRegistry` if they want to sort by name even within type by certain criteria eg sort by name on object fields but not directives say.    weird but now possible

### CODE REVIEW

**Comment by bbakerman** [comment]
This is a breaking change because `graphql.schema.GraphqlTypeComparatorRegistry` got renamed from the previous schema printer named code.  Let the same but moved and renamed.

SchemaPrinter BTW is still in name order!

**Comment by marceloverdijk** [comment]
> Since all of the maps / lists are not LinkedHashMap the insertion order will be stable.

@bbakerman I guess you meant: _Since all of the maps / lists are **now** LinkedHashMap the insertion order will be stable._

**Comment by bbakerman** [comment]
yes I did mean that.  Updated the comment

**Comment by marceloverdijk** [comment]
Looking forward to `13.0`. Any ETA on release? Like weeks, months?

File: src/main/java/graphql/schema/GraphqlTypeBuilder.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+import graphql.Internal;
+
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map;
+
+import static graphql.util.FpKit.valuesToList;
+
+@Internal
+abstract class GraphqlTypeBuilder {
```
Comment (andimarek) [inline src/main/java/graphql/schema/GraphqlTypeBuilder.java]: lets make it public

**Pull request merged**